compiler/src/typed/env.re

```diff
@@ -848,8 +848,10 @@ let acknowledge_pers_struct = (check, {Persistent_signature.filename, cmi}) => {
 
 let find_pers_struct = (~loc, check, filepath) => {
   switch (Hashtbl.find(persistent_structures, filepath)) {
-  | Some(ps) => ps
-  | None => raise(Not_found)
+  | Some(ps) =>
+    failwith("find_pers_struct unexpectedly found persistent_structures for .gr filepath " ++ filepath ++ ". Please open an issue.");
```

Comment:
Idea here is that `persistent_structures` should _never_ have a `filepath` entry in it because it is a `.gr` file whereas this map should only have `.gro` files in it.

To be honest I was not confident enough that some code path does not trigger this so rather than deleting this code entirely I just included these safeguards which would warrant additional investigation.